server/controllers/users.js

```diff
@@ -309,22 +312,39 @@ module.exports = (app) => {
           })
           .catch(next);
       } else if (req.query.profile) {
-        var groups;
-        req.user
-        .getGroups()
-        .then(results => {
-          groups = results.map(g => Object.assign(g.minimal, { role: g.UserGroup.role, createdAt: g.UserGroup.createdAt }));
-          return groups;
+        var groupInfoArray = []
+        req.user.getGroups()
+        .then(groups => {
+          return Promise.all(groups.map(group => {
+            return Promise.all([
+              group.getYearlyIncome(),
+              new Promise((resolve, reject) => {
+                groupsController.getUsers({group: group}, {send: resolve}, reject);
+              })
+            ])
+            .then(values => {
+              var groupInfo = group.info;
+              groupInfo.yearlyIncome = values[0];
+              var usersByRole = groupBy(values[1], 'role');
+              groupInfo.hosts = usersByRole[roles.HOST] || [];
```

Comment:
No, hosts are not needed I will remove that line

You can find see the component here: https://github.com/OpenCollective/opencollective-website/pull/319